Sources/Rosetta/PlayMode/Tasks/SimpleTasks/RandomCardTask.cpp

```diff
@@ -52,106 +52,66 @@ std::vector<Card*> RandomCardTask::GetCardList(Entity* source,
 {
     std::vector<Card*> result;
 
-    if (cardClass == CardClass::INVALID)
+    const auto cards = GetCardList(source, cardClass);
+    for (const auto& card : cards)
     {
-        const auto cards = source->game->GetFormatType() == FormatType::STANDARD
-                               ? Cards::GetAllStandardCards()
-                               : Cards::GetAllWildCards();
-
-        for (const auto& card : cards)
+        if ((cardType == CardType::INVALID ||
+             cardType == card->GetCardType()) &&
+            (race == Race::INVALID || race == card->GetRace()) &&
+            (rarity == Rarity::INVALID || rarity == card->GetRarity()))
         {
-            if ((cardType == CardType::INVALID ||
-                 cardType == card->GetCardType()) &&
-                (race == Race::INVALID || race == card->GetRace()) &&
-                (rarity == Rarity::INVALID || rarity == card->GetRarity()))
-            {
-                bool check = true;
+            bool check = true;
 
-                for (auto& tag : tags)
+            for (auto& tag : tags)
+            {
+                if (!card->HasGameTag(tag.first) ||
+                    card->gameTags[tag.first] != tag.second)
                 {
-                    if (!card->HasGameTag(tag.first) ||
-                        card->gameTags[tag.first] != tag.second)
-                    {
-                        check = false;
-                        break;
-                    }
+                    check = false;
+                    break;
                 }
 
-                if (check)
+                if (cardClass == CardClass::ANOTHER_CLASS &&
+                    source->player->GetHero()->card->GetCardClass() != cardClass)
                 {
-                    result.emplace_back(card);
+                    check = false;
```

Comment:
Add break statement.
```suggestion
                    check = false;
                    break;
```

Reply:
Your suggestion is also right however actually it seems another issue which `if` statement is in incorrect code block. Actually, in my intention, it should go out from the tags `for` statement, because there is no reason to check *it is another class card* for each filter tag. Then the code will be like below and it doesn't need `break` statement anymore, does it?

```cpp
for (auto& tag : tags)
{
    if (!card->HasGameTag(tag.first) ||
        card->gameTags[tag.first] != tag.second)
    {
        check = false;
        break;
    }
}

if (cardClass == CardClass::ANOTHER_CLASS &&
    source->player->GetHero()->card->GetCardClass() != cardClass)
{
    check = false;
}
```

Reply:
Oh, yes. This code seems right. 😀